docs/go/core/bzlmod.md

```diff
@@ -0,0 +1,180 @@
+# Go with Bzlmod
+
+This document describes how to use rules_go and Gazelle with Bazel's new external dependency subsystem [Bzlmod](https://bazel.build/external/overview#bzlmod), which is meant to replace `WORKSPACE` files eventually.
+Usages of rules_go and Gazelle in `BUILD` files are not affected by this, refer to the existing documentation on rules and configuration options for them.
+
+## Setup
+
+Add the following lines to your `MODULE.bazel` file:
+
+```starlark
+bazel_dep(name = "rules_go", version = "0.39.1")
+bazel_dep(name = "gazelle", version = "0.29.0")
+```
+
+The latest versions are always listed on https://registry.bazel.build/.
+
+If you have WORKSPACE dependencies that reference rules_go and/or Gazelle, you can still use the legacy repository names for the two repositories:
+
+```starlark
+bazel_dep(name = "rules_go", version = "0.39.1", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "gazelle", version = "0.29.0", repo_name = "bazel_gazelle")
+```
+
+## Registering Go SDKs
+
+rules_go automatically downloads and registers a recent Go SDK, so unless a particular version is required, no manual steps are required.
+
+To register a particular version of the Go SDK, use the `go_sdk` module extension:
+
+```starlark
+go_sdk = use_extension("@rules_go//go:extensions.bzl", "go_sdk")
+
+# Download an SDK for the host OS & architecture.
+go_sdk.download(version = "1.20.3")
+
+# Download an SDK for a fixed OS/architecture, e.g. for remote execution.
+go_sdk.download(
+    version = "1.20.3",
+    goarch = "amd64",
+    goos = "linux",
+)
+
+# Register the Go SDK installed on the host.
```

Comment:
Assuming that people usually only want one of these three options:
```suggestion
# Alternately, register the Go SDK installed on the host.
```
I see that you mention `go_cross_binary` below, so maybe people _will_ want to register several SDKs together. If they do, how will they identify each of them later? Should we show use of the "name" attribute here?

Reply:
`go_cross_binary` can change the platform and the required SDK version. We generally want people to specify what they want and receive a matching SDK without caring about which particular one they want. If they really want to pin a particular one, they should use `name` and `--extra_toolchains`, but that should be very rare.